pkg/sriov/sriov.go

```diff
@@ -91,7 +92,14 @@ func (s *sriovManager) SetupVF(conf *sriovtypes.NetConf, podifName string, cid s
 		// Save the original effective MAC address before overriding it
 		conf.OrigVfState.EffectiveMAC = linkObj.Attrs().HardwareAddr.String()
 
-		if err = s.nLink.LinkSetHardwareAddr(linkObj, hwaddr); err != nil {
+		// Retry up to 5 times; wait 200 milliseconds between retries
+		for retry := 0; retry < 5; retry++ {
```

Comment:
perhaps we can borrow this logic [1][2] and use it here and below WDYT ?

[1] https://github.com/kubernetes/apimachinery/blob/fd8a60496be5e4ce19d586738adb48ac6fa66ef9/pkg/util/wait/wait.go#L421
[2] https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait

Reply:
by borrow i mean copy :) in order not to drag in dependency on k8s

Reply:
I considered a back-off mechanism. In the end, it takes just up to a few milliseconds.
Do we really need one and is it worth maintaining it in our codebase or adding a dependency?

Reply:
at least not to have this code duplicated if you dont think this optimization is needed.

so some function that will do the retry logic would be great.

Reply:
Done.